### ISSUE TITLE
Remove h5py

### ISSUE THREAD

**Issue by bcmaton** [TRIQS/triqs#270]
Hi, 
I tried installing TRIQS on MacOS. I almost finished the installation but with problem of hdf5. I cannot pass the test with messages. 
The following tests FAILED:
	132 - gf_init (Failed)
	133 - gf_base_op (Failed)
	135 - pade (Failed)
	140 - wrap_a (Failed)
Errors while running CTest
make: *** [test] Error 8

cat test/pytriqs/base/gf_init_output.err
Traceback (most recent call last):
  File "<stdin>", line 33, in <module>
  File "/Users/kun/triqs.build/pytriqs/archive/hdf_archive.py", line 183, in __setitem__
    val.__write_hdf5__(self._group,key)
TypeError: Internal error : invalid id from h5py !

I found that this maybe the problem of h5py version and hdf5 version.
I tried brew install hdf5. But this is version 1.8.15 while h5py version is 1.8.13. Thus I installed hdf5 1.8.13 by myself with ./configure --enable=production. I cmake and make without any error. 

Do you have any suggestions to solve this issue?

P.S. the installation instructions is not complete with missing GMP. Thanks a lot for your attention.

**Comment by parcollet** [comment]
Hi, 
Yes, I have encountered the same problem recently.
A somewhat simpler solution is to upgrade h5py instead of hdf5.
The problem is due to OS X: because brew is changing all the time, one can easily end up with an hdf5 and an h5py with different versions, which is a nonsense.
It may also happen in a machine with multiple hdf5 modules.
The only real clean solution is to remove h5py from our used libraries.
We use it only as a basic layer in HDFArchive, so it means only to wrap a couple of functions from C.
To be discussed at a next meeting.
cc @mferrero , @krivenko, @tayral 

PS : I fixed the doc for GMP, thanks.
